compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3563,7 +3563,7 @@ public void visit(BLangInvocation.BLangActionInvocation aInv, AnalyzerData data)
     public void visit(BLangInvocation.BLangResourceAccessInvocation resourceAccessInvocation, AnalyzerData data) {
         // Find the lhs expression type
         checkExpr(resourceAccessInvocation.expr, data);
-        BType lhsExprType = resourceAccessInvocation.expr.getBType();
+        BType lhsExprType = Types.getReferredType(resourceAccessInvocation.expr.getBType());
```

Comment:
the error should have the `BReferenceType`.  Use a temp var for getting the referred type instead and check for the type tags